File: app/src/main/java/in/completecourse/questionbank/EditProfileActivity.java
package in.completecourse.questionbank;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.Spinner;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.FirebaseFirestore;

import java.util.HashMap;
import java.util.Map;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import in.completecourse.questionbank.adapter.SpinAdapterNew;

public class EditProfileActivity extends AppCompatActivity {
    @BindView(R.id.classSelection)
    Spinner spinner;
    @BindView(R.id.edTv_username)
    EditText edTvUsername;
    @BindView(R.id.edTv_school)
    EditText edTvSchool;
    @BindView(R.id.edTv_email)
    EditText edTvEmail;
    @BindView(R.id.edTv_contact)
    EditText edTvContact;
    @BindView(R.id.progressBar)
    ProgressBar progressBar;
    private FirebaseFirestore db;
    private FirebaseAuth mAuth;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_profile);


        ButterKnife.bind(this);

        db = FirebaseFirestore.getInstance();
        mAuth = FirebaseAuth.getInstance();

        String[] userClassArray = new String[]{"Choose Class", "9", "10", "11", "12"};
        SpinAdapterNew spinAdapter = new SpinAdapterNew(this, R.layout.spinner_row, userClassArray);

        spinner.setAdapter(spinAdapter);

        getUserProfile(mAuth.getUid());

    }

    @OnClick(R.id.btn_done)
    void done(){
        String username;
        String userClass = null;
        String userEmail;
        String userSchool;
        String contact;

        username = edTvUsername.getText().toString();
        userEmail = edTvEmail.getText().toString();
        userSchool = edTvSchool.getText().toString();
        contact = edTvContact.getText().toString();

        if (userEmail.equals("Not Provided")) userEmail = "";
        if (userSchool.equals("Not Provided")) userSchool = "";
        if (contact.equals("Not Provided")) contact = "";

        if (spinner.getSelectedItemPosition() == 0) userClass = "";
        else if (spinner.getSelectedItemPosition() == 1) userClass = "9th";
        else if (spinner.getSelectedItemPosition() == 2) userClass = "10th";
        else if (spinner.getSelectedItemPosition() == 3) userClass = "11th";
        else if (spinner.getSelectedItemPosition() == 4) userClass = "12th";
        if (userClass != null) {
            if (!username.isEmpty() && !userEmail.isEmpty() && !userClass.isEmpty() &&
                    !userSchool.isEmpty() && !contact.isEmpty()) {
                updateProfile(username, userClass, userEmail, userSchool, contact);
            } else Toast.makeText(this, "Please fill all the details", Toast.LENGTH_SHORT).show();
        }
    }

    private void getUserProfile(String userId){
        db.collection("qb_users").document(userId).get().addOnCompleteListener(task -> progressBar.setVisibility(View.GONE)).addOnSuccessListener(documentSnapshot -> {

            edTvUsername.setText(documentSnapshot.getString("name"));

            if (documentSnapshot.getString("email") == null) edTvEmail.setText("Not Provided");
            else edTvEmail.setText(documentSnapshot.getString("email"));

            String userClass = documentSnapshot.getString("class");
            if (userClass == null) spinner.setSelection(0);
            else if (userClass.equals("9th")) spinner.setSelection(1);
            else if (userClass.equals("10th")) spinner.setSelection(2);
            else if (userClass.equals("11th")) spinner.setSelection(3);
            else if (userClass.equals("12th")) spinner.setSelection(4);

            if (documentSnapshot.getString("school") == null) edTvSchool.setText("Not Provided");
            else edTvSchool.setText(documentSnapshot.getString("school"));

            if (documentSnapshot.getString("phone") == null) edTvContact.setText("--");
            else edTvContact.setText(documentSnapshot.getString("phone"));

        }).addOnFailureListener(e -> Toast.makeText(this, e.getMessage(), Toast.LENGTH_SHORT).show());

    }

    private void updateProfile(String username, String userClass, String userEmail, String userSchool, String contact) {
        progressBar.setVisibility(View.VISIBLE);
        Map<String, Object> data = new HashMap<>();
        data.put("name", username);
        data.put("class", userClass);
        data.put("email", userEmail);
        data.put("school", userSchool);
        data.put("phone", contact);
        if (mAuth.getUid() != null) {
            db.collection("qb_users")
                    .document(mAuth.getUid()).set(data)
                    .addOnSuccessListener(aVoid -> {
                        Toast.makeText(this, "Profile updated successfully", Toast.LENGTH_SHORT).show();
                        finish();
                    })
                    .addOnFailureListener(e -> Toast.makeText(this, e.getMessage(), Toast.LENGTH_SHORT).show())
                    .addOnCompleteListener(task -> progressBar.setVisibility(View.GONE));
        }

    }
}

File: app/src/main/java/in/completecourse/questionbank/EditProfileActivity.kt
package `in`.completecourse.questionbank

import `in`.completecourse.questionbank.adapter.SpinAdapterNew
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.ProgressBar
import android.widget.Spinner
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import com.google.android.gms.tasks.Task
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.DocumentSnapshot
import com.google.firebase.firestore.FirebaseFirestore
import kotlinx.android.synthetic.main.activity_edit_profile.*
import java.util.*

class EditProfileActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit_profile)

        val userClassArray = arrayOf("Choose Class", "9", "10", "11", "12")
        val spinAdapter = SpinAdapterNew(this, R.layout.spinner_row, userClassArray)
        classSelection.adapter = spinAdapter
        getUserProfile(FirebaseAuth.getInstance().uid)

        btn_done.setOnClickListener{
            done()
        }
    }

    fun done() {
        val username: String
        var userClass: String? = null
        var userEmail: String
        var userSchool: String
        var contact: String

        username = edTv_username.text.toString()
        userEmail = edTv_email.text.toString()
        userSchool = edTv_school.text.toString()
        contact = edTv_contact.text.toString()

        if (userEmail == "Not Provided") userEmail = ""
        if (userSchool == "Not Provided") userSchool = ""
        if (contact == "Not Provided") contact = ""
        if (classSelection.selectedItemPosition == 0)
            userClass = ""

        else if (classSelection.selectedItemPosition == 1)
            userClass = "9th"
        else if (classSelection.selectedItemPosition == 2)
            userClass = "10th"
        else if (classSelection.selectedItemPosition == 3)
            userClass = "11th"
        else if (classSelection.selectedItemPosition == 4)
            userClass = "12th"

        if (userClass != null) {
            if (!username.isEmpty() && !userEmail.isEmpty() && !userClass.isEmpty() && !userSchool.isEmpty() && !contact.isEmpty()) {
                updateProfile(username, userClass, userEmail, userSchool, contact)
            } else Toast.makeText(this, "Please fill all the details", Toast.LENGTH_SHORT).show()
        }
    }

    private fun getUserProfile(userId: String?) {
        FirebaseFirestore.getInstance().collection("qb_users").document(userId!!).get().addOnCompleteListener { task: Task<DocumentSnapshot?>? -> progressBar!!.visibility = View.GONE }.addOnSuccessListener { documentSnapshot: DocumentSnapshot ->
            edTv_username.setText(documentSnapshot.getString("name"))
            if (documentSnapshot.getString("email") == null)
                edTv_email.setText("Not Provided")
            else
                edTv_email.setText(documentSnapshot.getString("email"))
            val userClass = documentSnapshot.getString("class")
            if (userClass == null)
                classSelection.setSelection(0)
            else if (userClass == "9th")
                classSelection.setSelection(1)
            else if (userClass == "10th")
                classSelection.setSelection(2)
            else if (userClass == "11th")
                classSelection.setSelection(3)
            else if (userClass == "12th")
                classSelection.setSelection(4)
            if (documentSnapshot.getString("school") == null)
                edTv_school.setText("Not Provided")
            else
                edTv_school.setText(documentSnapshot.getString("school"))
            if (documentSnapshot.getString("phone") == null)
                edTv_contact.setText("--")
            else
                edTv_contact.setText(documentSnapshot.getString("phone"))
        }.addOnFailureListener { e: Exception -> Toast.makeText(this, e.message, Toast.LENGTH_SHORT).show() }
    }

    private fun updateProfile(username: String, userClass: String, userEmail: String, userSchool: String, contact: String) {
        progressBar!!.visibility = View.VISIBLE
        val data: MutableMap<String, Any> = HashMap()
        data["name"] = username
        data["class"] = userClass
        data["email"] = userEmail
        data["school"] = userSchool
        data["phone"] = contact
        if (FirebaseAuth.getInstance().uid != null) {
            FirebaseFirestore.getInstance().collection("qb_users")
                    .document(FirebaseAuth.getInstance().uid!!).set(data)
                    .addOnSuccessListener { aVoid: Void? ->
                        Toast.makeText(this, "Profile updated successfully", Toast.LENGTH_SHORT).show()
                        finish()
                    }
                    .addOnFailureListener { e: Exception -> Toast.makeText(this, e.message, Toast.LENGTH_SHORT).show() }
                    .addOnCompleteListener { task: Task<Void?>? -> progressBar!!.visibility = View.GONE }
        }
    }
}